contracts/strategies/SnowballProxyStrategyForLPV1.sol

```diff
@@ -0,0 +1,235 @@
+        proxy = ISnowballProxy(_snowballProxy);
+    }
+
+    function setAllowances() public override onlyOwner {
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "SnowballStrategyForLPV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(proxy), amount));
+        uint lpAmountDeposited = _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(lpAmountDeposited));
+        totalDeposits = totalDeposits.add(lpAmountDeposited);
+        emit Deposit(account, lpAmountDeposited);
+    }
+
+    function _stakeDepositTokens(uint amount) private returns (uint) {
+        require(amount > 0, "SnowballStrategyForLPV1::_stakeDepositTokens");
+        return proxy.deposit(stakingContract, snowGlobe, address(depositToken));
+    }
+
+    function withdraw(uint amount) external override {
+        uint depositTokenAmount = getDepositTokensForShares(amount);
+        if (depositTokenAmount > 0) {
+            uint amountReceived = _withdrawDepositTokens(depositTokenAmount);
+            _safeTransfer(address(depositToken), msg.sender, amountReceived);
+            _burn(msg.sender, amount);
+            totalDeposits = totalDeposits.sub(depositTokenAmount);
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private returns (uint) {
+        require(amount > 0, "SnowballStrategyForLPV1::_withdrawDepositTokens");
+        return proxy.withdraw(stakingContract, snowGlobe, address(depositToken), amount);
+    }
+
+    function reinvest() external override onlyEOA {
+        uint unclaimedRewards = checkReward();
+        require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "SnowballStrategyForLPV1::reinvest");
+        _reinvest(_convertRewardIntoWAVAX(unclaimedRewards));
+    }
+
+    function _convertRewardIntoWAVAX(uint pendingReward) private returns (uint) {
+        proxy.claimReward(stakingContract);
+        DexLibrary.swap(
+            pendingReward,
+            address(rewardToken), address(WAVAX),
+            swapPairWAVAXSnob
+        );
+        return WAVAX.balanceOf(address(this));
```

Comment:
dexLibrary.swap returns the output amount, which is cheaper than a balanceOf here

Reply:
fixed